src/top/run/metrics_util.py

```diff
@@ -0,0 +1,280 @@
+"""
+Util classes for computing evaluation metrics.(HitMiss, Accuracy, AveragePrecision, IoU)
+Reference: https://github.com/google-research-datasets/Objectron/blob/master/objectron/dataset/metrics.py
+"""
+
+import numpy as np
+import scipy.spatial as sp
+import src.top.run.box_generator as Box
+
+
+class HitMiss(object):
+    """Class for recording hits and misses of detection results."""
+
+    def __init__(self, thresholds):
+        self.thresholds = thresholds
+        self.size = thresholds.shape[0]
+        self.hit = np.zeros(self.size, dtype=np.float)
+        self.miss = np.zeros(self.size, dtype=np.float)
+
+    def reset(self):
+        self.hit = np.zeros(self.size, dtype=np.float)
+        self.miss = np.zeros(self.size, dtype=np.float)
+
+    def record_hit_miss(self, metric, greater=True):
+        """Records the hit or miss for the object based on the metric threshold."""
+        # NOTE(Jiyong): greater True -> find greater than threshold / False -> find worse than threshold
```

Comment:
This is a comment about the argument - as such, it should be included in the function docstring.